includes/class-wcpdf-main.php

```diff
@@ -1582,25 +1582,31 @@ public function wc_webhook_trigger( $document, $order ) {
 	 * @return void
 	 */
 	public function display_due_date( string $document_type, $order ): void {
-		if ( 'invoice' !== $document_type || empty( $order ) ) {
+		if ( empty( $order ) ) {
 			return;
 		}
 
-		$invoice = wcpdf_get_invoice( $order );
+		$allowed_document_types = apply_filters( 'wpo_wcpdf_due_date_allowed_document_types', array( 'invoice' ), $order, $document_type );
```

Comment:
Looking into this, maybe we could avoid this filter only by checking if the `get_due_date` is callable?

Reply:
Or we can make it possible to get their custom function name, and we call that instead. And then it's ok to keep the filter.
What do you think?

Reply:
I'm not sure if I understand.

Reply:
I meant something like this:
```php
		$due_date_timestamp = 0;
		$due_date_function  = apply_filters( 'wpo_wcpdf_getting_due_date_function_name', '', $order, $document_type );

		if ( ! empty( $due_date_function ) ) {
			if ( is_callable( $due_date_function ) ) {
				$due_date_timestamp = call_user_func( $due_date_function );
			}
		} else {
			if ( is_callable( array( $document, 'get_due_date' ) ) ) {
				$due_date_timestamp = $document->get_due_date();
			}
		}
```

With this, we can create a callable function for other documents with a snippet if users request.

Reply:
Why we would use custom functions? Do you have a scenario when that could happen? I think if we add to the abstract class, we could use in every document if the setting exist there right?

Reply:
The only scenario that came to my mind is when a user wants to add the due date for other types of documents for any reason. In that case, allowing to call a custom function can easily make it possible.

> Maybe we could add to the abstract document class and use in the documents we want when the setting exists?

Do you think we will add the due date for other documents? I think it'll be only for the invoice, so it may not be necessary to add it there. Not sure!

Reply:
> Do you think we will add the due date for other documents? I think it'll be only for the invoice, so it may not be necessary to add it there. Not sure!

I can see this being used in the Proposal or similar documents like Quotes.

Reply:
Ok then, I'll add the `get_due_date()` to the abstract document class and remove `$allowed_document_types`.